pallets/xyk/src/lib.rs

```diff
@@ -955,9 +955,7 @@ impl<T: Trait> XykFunctionsTrait<T::AccountId> for Module<T> {
         ensure!(first_asset_id != second_asset_id, Error::<T>::SameAsset,);
 
         // Liquidity token amount calculation
-        let initial_liquidity = first_asset_amount
-            .checked_add(second_asset_amount)
-            .ok_or_else(|| DispatchError::from(Error::<T>::MathOverflow))?;
+        let initial_liquidity = first_asset_amount;
```

Comment:
@stanieltron , 
I think we have bunch of calculations based on this value. Are you sure that this will work? I'm not 100% confident, but some calculations assume that this liquidity amounts are nX+nY. Am I wrong?

ie:
burn _liquidity, L656,

`       
        let total_liquidity_assets: Balance =
            <T as Trait>::Currency::total_issuance(liquidity_asset_id.into()).into();
 let first_asset_amount = multiply_by_rational(
            first_asset_reserve,
            liquidity_asset_amount,
            total_liquidity_assets,
        )`